app/products/calls/connection/websocket_event_handlers.ts

```diff
@@ -228,3 +232,32 @@ export const handleHostLowerHand = async (serverUrl: string, msg: WebSocketMessa
 
     unraiseHand();
 };
+
+export const handleHostRemoved = async (serverUrl: string, msg: WebSocketMessage<HostControlsMsgData>) => {
+    const currentCall = getCurrentCall();
+    if (currentCall?.serverUrl !== serverUrl ||
+        currentCall?.channelId !== msg.data.channel_id ||
+        currentCall?.mySessionId !== msg.data.session_id) {
+        // wasn't the current call, check previous call:
+        const previousCall = getPreviousCall();
+        if (previousCall?.serverUrl !== serverUrl ||
+            previousCall?.channelId !== msg.data.channel_id ||
+            previousCall?.mySessionId !== msg.data.session_id) {
+            return;
```

Comment:
just to be clear.. user can only be in one call at a time, this previous call is just in case of a race condition where the current call is set to something else?

Reply:
Yes, race condition for receiving the `host_removed` after the current call has ended. I could create one by delaying the ws event by 10ms on the server, so I wanted to be safe and take care of it on the client side.

Reply:
Do we really need to handle this edge case if the call has ended? Is the main purpose of this to show the alert?

Reply:
@streamer45 The edge case is not really an edge case, it's almost like the expected case with a bit of delay or load on the server. It's rare but not unexpected that the user would receive this message after the call has ended. Does that make sense?

Reply:
Is it likely that a host would remove someone from a call one second before ending it for everyone else (or leaving I suppose, then it's a 1-1)?

And if that happens, does it matter if we show the message or not?

Reply:
Maybe there's a miscommunication, or I'm misunderstanding what you're getting at. What I mean: the race could happen if the mobile client's rtc session is closed, it's been processed (the currentCall is set to null), and then the `hostRemoved` ws event arrives. Maybe it would never happen in practice, I don't know. This just makes sure if that did happen, the client knows that the message is related to the call they were just removed from. Otherwise there's no record in the state that the user was in a call at all (other than `currentCall` we don't keep a record of previous calls -- now we do).

Reply:
I see, so by call ended you meant the client closing.

I guess if we handled this as an error case on the client we wouldn't have had to deal with a race like this.

Reply:
@streamer45 But the session closing is not an error case, right?

Reply:
Nope, sending a `wsError` is though. Anyway, just thinking out loud because I realized you had to add this previous call handling just for this.

Reply:
@streamer45  I think we might have the same problem -- we have to send to the userID, but we have to know what the user's sessionID for the closed call was. If we send the error before, we'll have an opposite race problem with bad UX (error modal over a running call).

Reply:
Wouldn't it be simpler to slightly delay closing the RTC channel from the server? That's a fail-safe but we'd like to do it cleanly from the client when possible.

Reply:
Done on the server in: https://github.com/mattermost/mattermost-plugin-calls/commit/352b3c0c6a7a070ffcc58837e51c334d52cb5654 , and removed previousCall code here. Thanks